lm_eval/tasks/ja/jaqket_v2.py

```diff
@@ -0,0 +1,243 @@
+            ),  # Exact match (the normalized answer exactly match the gold answer)
+            "f1": (
+                predictions,
+                references,
+            ),  # The F-score of predicted tokens versus the gold answer
+        }
+
+
+    def aggregation(self):
+        return {
+            "exact_match": partial(
+                self._squad_agg, "exact_match"
+            ),  # Exact match (the normalized answer exactly match the gold answer)
+            "f1": partial(
+                self._squad_agg, "f1"
+            ),  # The F-score of predicted tokens versus the gold answer
+        }
+    
+    def higher_is_better(self):
+        return {
+            "exact_match": True,  # Exact match (the normalized answer exactly match the gold answer)
+            "f1": True,  # The F-score of predicted tokens versus the gold answer
+        }
+
+    def _squad_metric(self, predictions, references):
+        return self.jasqaud_metric.compute(predictions=predictions, references=references)
+
+
+    def _squad_agg(self, key, item):
+        predictions, references = zip(*item)
+        return self._squad_metric(predictions=predictions, references=references)[key]
+
+class JAQKETV2WithFintanPrompt(JAQKETV2):
+    """
+    prompt template is taken from [ChatGPT vs BERT: どちらが日本語をより理解できるのか?](https://fintan.jp/page/9126/)
+    """
+    PROMPT_VERSION = 0.2
+    DESCRIPTION = "質問に対する回答を文章から一言で抽出してください。回答は名詞で答えてください。\n\n"
+    SEP = "\n"
+    TOP_K_LIMIT = _TOP_K_LIMIT
+    def doc_to_text(self, doc):
+        text = f"{self.SEP}".join([ctx for ctx in doc["ctxs"]["text"][:self.TOP_K_LIMIT]])
+        return (
+            "文章:"
+            + text
+            + f"{self.SEP}"
+            + "質問:"
+            + doc["question"]
+            + f"{self.SEP}"
+            + "回答:"
+        )
+    
+
+class JAQKETV2WithJAAlpacaPrompt(JAQKETV2):
+    """
+    This prompt format was inspired by the below data in fujiki/japanese_alpaca_data. 
+    ```
+    {
+        'instruction': '与えられた文脈に最も適した文を選択してください。', 
+        'input': '文脈：あなたは親友と現在の仕事の状況について話しています。\nA）私にはあまり選択肢がありません。\nB）他に選択肢がありません。\nC）私には本当に決断する必要がありません。', 
+        'output': 'A) 私には多くの選択肢がありません。'
+    }
+    ```
+    Reference:
+    - data: https://huggingface.co/datasets/fujiki/japanese_alpaca_data
+    - code: https://github.com/Stability-AI/gpt-neox/blob/c130a4edc1120dccec8f02a34eb60d3e8f484cd3/finetune/finetune_base_ja.py#LL118C23-L127C11
+    """
+    PROMPT_VERSION = 0.3
+    DESCRIPTION = "以下は、タスクを説明する指示と、文脈のある入力の組み合わせです。要求を適切に満たす応答を書きなさい。\n\n\n"
+    INSTRUCTION = "与えられた文脈から、質問に対する答えを抜き出してください。\n\n"
+    # SEP = ""
+    TOP_K_LIMIT = _TOP_K_LIMIT
+    def doc_to_text(self, doc):
+        """
+        以下は、タスクを説明する指示と、文脈のある入力の組み合わせです。要求を適切に満たす応答を書きなさい。
+
+        ### 指示: 
+        {instruction}
+
+        ### 入力: 
+        {input}
+
+        ### 応答: 
+        {response}
+        """
+        text = "".join([ctx for ctx in doc["ctxs"]["text"][:self.TOP_K_LIMIT]])
```

Comment:
I think you can also join by `self.SEP` here as well. 
```suggestion
        text = f"{self.SEP}.join([ctx for ctx in doc["ctxs"]["text"][:self.TOP_K_LIMIT]])
```

Reply:
thank you for pointing this out. I think we can use `SEP` too.